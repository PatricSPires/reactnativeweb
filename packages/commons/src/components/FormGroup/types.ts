type Input = {
  name?: string
  value?: string
  onFocus?: () => void
  onBlur: () => void
}

export type FormGroupProps = {
  name: string
  tag: unknown
  label?: string
  width?: string
  labelHint?: string
  activeAppearance?: string
  className?: string
  placeholder?: string
  helperText?: string
  meta?: {
    active?: boolean
    touched?: boolean
    error?: string
    valid?: boolean
  }
  input?: Input
  iconPosition?: 'left' | 'right'
  icon?: React.ReactNode
  iconColor?: 'error' | 'warning' | 'success'
  isInvalid?: boolean
  passwordField?: boolean
  marginBottom: 1 | 2 | 3 | 4 | 5 | 6 | 7
}

export type FormGroupWrapperProps = Pick<
  FormGroupProps,
  'width' | 'marginBottom'
>
