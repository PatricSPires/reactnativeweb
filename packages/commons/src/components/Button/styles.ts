import styled, { css } from 'styled-components/native'
import { ThemeProps } from '../../styles/customThemes/types'

import { ButtonTextProps, ButtonWrapperProps } from './types'
import { getButtonAppearance, getTextAppearance } from './appearance'

export const ButtonWrapper = styled.TouchableOpacity<
  ButtonWrapperProps & ThemeProps
>`
  ${({ alignIcon, alignItems, theme }) => css`
    display: flex;
    align-items: center;
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    border: none;
    padding: ${theme.spacings[2]};
    flex-direction: ${alignIcon === 'left' ? 'row-reverse' : 'row'};
    justify-content: ${alignItems === 'center' ? 'center' : 'space-between'};
    ${getButtonAppearance}
  `}
`

export const ButtonText = styled.Text<ButtonTextProps>`
  ${getTextAppearance}
`
