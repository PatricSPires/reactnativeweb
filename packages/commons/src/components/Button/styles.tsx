import React from 'react'
import styled, { css } from 'styled-components'
import { Platform, TouchableOpacity, Text } from 'react-native'
import { ButtonProps } from './types'
import { buttonAppearances, textAppearances } from './appearance'

type ButtonComponentProps = Pick<
  ButtonProps,
  | 'appearance'
  | 'marginBottom'
  | 'width'
  | 'alignIcon'
  | 'alignItems'
  | 'className'
>
type TextComponentProps = Pick<ButtonProps, 'alignText' | 'appearance'>

type ButtonComponent1Props = {
  className?: string
  children: JSX.Element
  onPress: () => void
}

const Button = styled(TouchableOpacity)`
  ${({ appearance }: ButtonComponentProps) => css`
    border: none;
    display: flex;
    flex-direction: ${({ alignIcon }: ButtonComponentProps) =>
      alignIcon === 'left' ? 'row-reverse' : 'row'};
    align-items: center;
    justify-content: ${({ alignItems }: ButtonComponentProps) =>
      alignItems === 'center' ? 'center' : 'space-between'};
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    padding: 16px;

    ${Platform.OS === 'web' && `cursor: pointer;`}
    ${!!appearance && buttonAppearances[appearance]()};
  `}
`
const ButtonComponent = ({
  onPress,
  className,
  children,
  ...props
}: ButtonComponent1Props) => {
  // eslint-disable jsx-props-no-spreading
  return (
    <Button
      className={className}
      {...props}
      as={Platform.OS === 'web' ? 'button' : TouchableOpacity}
      onClick={onPress}
      onPress={onPress}
    >
      {children}
    </Button>
  )
}

ButtonComponent.defaultProps = {
  className: null,
}

export const TextComponent = styled(Text)<TextComponentProps>`
  ${({ appearance }: TextComponentProps) => css`
    ${!!appearance && textAppearances[appearance]()};
  `}
`
