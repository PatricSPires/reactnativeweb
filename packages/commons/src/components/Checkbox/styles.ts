import { Text, TouchableWithoutFeedback, View } from 'react-native'
import styled from 'styled-components/native'
import theme from '../../styles/theme'
import { CheckboxProps } from './types'

export const CheckBoxWrapper = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 2px;
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    width: 15px;
    height: 15px;
  }

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === false) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked === false && disabled === true) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
        background-color: transparent;
      `
    }
    return ``
  }}
`

export const CheckBoxCheck = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const CheckBoxText = styled(Text)`
  margin-left: 8px;
  color: ${(props: CheckboxProps) =>
    props.disabled
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`
