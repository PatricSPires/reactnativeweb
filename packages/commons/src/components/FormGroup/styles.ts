import styled from 'styled-components'
import { View, Text } from 'react-native'
import theme from '../../styles/theme'
import { FormGroupWrapperProps } from './types'

export const FormGroupWrapper = styled(View)<FormGroupWrapperProps>`
  display: flex;
  width: ${({ width }) => width || '100%'};
  margin-bottom: ${({ marginBottom }) => theme.spacings[marginBottom || 3]};

  Tag {
    margin: 8px 0 4px 0;
  }
`

export const Label = styled(Text)`
  color: ${(props) => props.theme.colors.font_color};
  font-size: ${theme.fontSize.label_01};
  margin-bottom: ${theme.spacings[1]};
  display: flex;
  align-items: center;
`

export const HelperText = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.supportColors.error};
  margin-top: ${theme.spacings[1]};
`
