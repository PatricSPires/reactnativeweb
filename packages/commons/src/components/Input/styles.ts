import styled, { css } from 'styled-components'
import {
  Text,
  TextInput as InputRN,
  View,
  TouchableWithoutFeedback,
} from 'react-native'
import { TextInputProps, IconProps, InputWrapperProps } from './types'
import theme from '../../styles/theme'

function getFocusStyles(isInvalid: boolean) {
  if (isInvalid) {
    return `
      outline: 2px solid ${theme.supportColors.error};
      border-bottom-color: transparent;
    `
  }
  return `
    outline: 2px solid ${theme.colors.primary_background};
    border-bottom-color: transparent;
  `
}

export const Wrapper = styled(View)`
  width: 100%;
`

export const InputWrapper = styled(View)<InputWrapperProps>`
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props) =>
    props.theme.colors.primary_container_background};
  border-bottom: 1px solid ${theme.colors.medium_contrast_border};

  ${({ isInvalid }) => css`
    outline: ${!!isInvalid && `2px solid ${theme.supportColors.error}`};
    border: ${!!isInvalid && 'none'};
  `}

  &:focus-within {
    ${({ isInvalid }) => getFocusStyles(isInvalid)}
  }
`

export const TextInput = styled(InputRN)<TextInputProps>`
  width: 100%;
  height: 40px;
  color: ${(props) => props.theme.colors.font_color};
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;

  order: ${({ iconPosition }) => (iconPosition === 'right' ? 0 : 1)};
  padding: ${({ iconPosition }) =>
    iconPosition === 'left' ? '0 10px 0 0' : '0 0 0 10px'};

  &:disabled {
    outline: none;
    border: none;
    color: ${(props) => props.theme.colors.font_color};
  }

  ::placeholder {
    color: ${(props) => props.theme.colors.font_color};
  }

  &:focus-within {
    color: ${(props) => props.theme.colors.font_color};
  }
`

export const HelperText = styled(Text)`
  font-size: 12px;
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: 12px;
  color: ${theme.supportColors.error};
`

export const Icon = styled(View)<IconProps>`
  order: ${({ iconPosition }) => (iconPosition === 'right' ? 1 : 0)};
  color: ${theme.colors.primary_background};
  margin: 0 10px;

  svg {
    width: 16px;
    ${({ iconColor }: IconProps) => css`
      color: ${theme.supportColors[iconColor]};
    `}
  }
`

export const PasswordIconWrapper = styled(TouchableWithoutFeedback)`
  width: 16px;
  height: 16px;
  color: ${(props) => props.theme.colors.font_color};
  margin: 0 10px;
`
