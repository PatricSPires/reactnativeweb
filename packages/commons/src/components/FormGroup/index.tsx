// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
// TODO: Tentamos achar uma forma de corrigir a tipagem da tag, mas, infelizmente falhamos, a final somos humanos.
import React from 'react'
import { FormGroupProps } from './types'
import { FormGroupWrapper, Label, HelperText, ErrorMessage } from './styles'
import Tooltip from '../Tooltip'

const FormGroup = ({
  tag: Tag,
  label,
  width,
  labelHint,
  placeholder,
  helperText,
  iconPosition,
  icon,
  iconColor,
  passwordField,
  marginBottom,
  meta = {},
  input = {
    onBlur: Function,
  },
  ...props
}: FormGroupProps) => {
  const isActive = meta.active
  const isInvalid = meta.error !== undefined && meta.touched && !isActive
  const isValid = meta.valid && meta.touched && !isActive

  return (
    <FormGroupWrapper width={width} marginBottom={marginBottom}>
      {!!label && (
        <Label>
          {label}
          {/* {!!labelHint && <Tooltip hint={labelHint} />} */}
        </Label>
      )}
      {!!helperText && <HelperText>{helperText}</HelperText>}
      <Tag
        {...input}
        {...props}
        placeholder={placeholder}
        active={isActive}
        isInvalid={isInvalid}
        isValid={isValid}
        iconPosition={iconPosition}
        icon={icon}
        iconColor={iconColor}
        passwordField={passwordField}
      />
      {!!isInvalid && <ErrorMessage>{meta.error}</ErrorMessage>}
    </FormGroupWrapper>
  )
}

FormGroup.defaultProps = {
  marginBottom: 3,
}

export default FormGroup
