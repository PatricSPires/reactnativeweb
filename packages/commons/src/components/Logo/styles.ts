import styled from 'styled-components/native'
import { ThemeProps } from '../../styles/types'

const LogoWrapper = styled.View`
  margin-bottom: ${({ theme }: ThemeProps) => theme.spacings[6]};
`
export default LogoWrapper
