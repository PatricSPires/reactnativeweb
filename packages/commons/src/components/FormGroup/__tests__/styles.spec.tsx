import React from 'react'
import 'jest-styled-components'
import { render } from '@testing-library/react-native'
import FormGroup from '../index'
import { FormGroupWrapper } from '../styles'

describe('FormGroup > Style', () => {
  it('should render the FormGroup with some styles', () => {
    const { toJSON } = render(<FormGroup tag="input" name="input" />)
    expect(toJSON()).toMatchSnapshot()
  })
})

describe('FormGroupWrapper > Style', () => {
  it('should render the FormGroupWrapper with width', () => {
    const { toJSON } = render(
      <FormGroupWrapper width="100px" marginBottom={2} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the FormGroupWrapper with margin bottom', () => {
    const { toJSON } = render(<FormGroupWrapper marginBottom={2} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
