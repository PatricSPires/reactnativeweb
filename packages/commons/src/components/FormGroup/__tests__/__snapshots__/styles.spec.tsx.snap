// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup > Style should render the FormGroup with some styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-bottom: 24px;
}

.c0 Tag {
  margin: 8px 0 4px 0;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
>
  <input
    isInvalid={false}
    name="input"
    onBlur={[Function]}
  />
</div>
`;

exports[`FormGroupWrapper > Style should render the FormGroupWrapper with margin bottom 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-bottom: 16px;
}

.c0 Tag {
  margin: 8px 0 4px 0;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
/>
`;

exports[`FormGroupWrapper > Style should render the FormGroupWrapper with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100px;
  margin-bottom: 16px;
}

.c0 Tag {
  margin: 8px 0 4px 0;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
/>
`;
